src/main/java/bio/terra/common/AclUtils.java

```diff
@@ -18,9 +18,13 @@ public class AclUtils {
   private static final int RETRIES = 15;
   private static final int MAX_WAIT_SECONDS = 30;
   private static final int INITIAL_WAIT_SECONDS = 2;
+  private static final Random RANDOM = new Random();
+
+  private AclUtils() {
+    throw new IllegalStateException("Utility classes are not meant to be instantiated");
```

Comment:
Did you get a warning using an empty constructor? I don't think you need to throw here.

Reply:
Sonar emitted a code smell and made this recommendation, but I was not sold on the value of this change.  I'm glad you commented on it because I was curious what others thought.

Reply:
I think adding a private constructor should be sufficient to remove the sonar warning (or at least it seems to work when I make this change locally).

It's not something we commonly do in DSP, but another way to write a utility (or singleton) class is to use an `enum`. I like that better since you don't a private constructor, or to declare the class `final` (since enums can't be subclassed). The odd thing is you need to add an empty instance list at the top. It would look like:
```
public enum AclUtils {
  ;
```